Project 4/Ali_Project4

```diff
@@ -0,0 +1,101 @@
+PREFIX ex: <http://example.org/ns/>
+PREFIX wiki: <http://dbpedia.org/resource/>
+PREFIX schema: <http://schema.org/>
+PREFIX rdf: <http://w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX owl: <http://w3.org/2002/07/owl#>
+PREFIX rdfs: <http://w3.org/2000/01/rdf-schema#>
+
+
+
+Ali's Kata 8 question: Find the area code used for phone lines in Seattle.
+
+SELECT ?o
+WHERE {dbr:Seattle dbo:areaCode ?o}
+
+--------------------------
+Ali’s Kata 7 question: SELECT queries are used for selecting bindings for variables, but you can also ask YES/NO questions that can be used to decide whether to include a specific section of the reporting software, or support software to make recommendations.
+To do so, we use the ASK keyword. For example, for the question “Did Elton John have an occupation?” we could ask
+
+ASK WHERE { dbr:Elton_John dbo:occupation ?any }  
+
+The answer would be True (yes). Your assignment is to ask a similar question about Aristotle by using FILTER.
+You must also have your answer as False (no).
+
+Solution:
+ASK WHERE { FILTER NOT EXISTS { dbr:Elton_John dbo:occupation ?any }}
+
+--------------------------
+Ali’s Kata 6 question: Write a query using the following prefixes to find data featuring information about SPARQL; mentions, has features of, or just has the keyword “SPARQL” appear in it. Make sure your code grabs the most amount of data that have something to do with SPARQL and 
+ 
+PREFIX ex: <http://example.org/ns/>
+PREFIX wiki: <http://dbpedia.org/resource/>
+PREFIX schema: <http://schema.org/>
+SELECT DISTINCT ?s ?p ?o 
+WHERE {
+    { 
+      ?s ex:hasFeature wiki:SPARQL . }
+    UNION {
+      ?s schema:mentions wiki:SPARQL . }
+    UNION {
+      ?s schema:keywords "SPARQL" . }
+}
+LIMIT 100
+
```

Comment:
I think this works, perhaps you could help the user by saying something more specific about the properties they need to use.  
Also, you use ex:hasFeature but you never mention ex: in the prefixes.
Also, do you think we would achieve the same result if we wrote 
WHERE {
    { 
      ?s ex:hasFeature wiki:SPARQL ;
           schema:mentions wiki:SPARQL ;
          schema:keywords "SPARQL" . 
}

Reply:
@giacomodecolle Your alternative query requires the same resource (?s) to have a value for 3 different properties at the same time. But maybe some resource has a value only for one of them. @ahasanzadeh13 But if we cannot guarantee that all resources have a value for the property "ex:hasFeature", then we should use OPTIONAL. So our query will look like:

WHERE {
   OPTIONAL 
   {?s ex:hasFeature wiki:SPARQL . }
    OPTIONAL 
   {?s schema:mentions wiki:SPARQL . }
    OPTIONAL 
   {?s schema:keywords "SPARQL" . }
}